src/auction.js

```diff
@@ -690,7 +690,7 @@ function getPreparedBidForAuction(bid, {index = auctionManager.index} = {}) {
   events.emit(CONSTANTS.EVENTS.BID_ADJUSTMENT, bid);
 
   // a publisher-defined renderer can be used to render bids
-  const adUnitRenderer = index.getAdUnit(bid).renderer;
+  const bidRenderer = index.getBidRequest(bid)?.renderer || index.getAdUnit(bid).renderer;
```

Comment:
Is the rendering order of priority of all four are present the following? 

Mediatype
Ad unit
Bid
Adapter

And if backup only is specified the order is:
Bid
Adapter
Media type
Ad unit

Correct?

Reply:
@fowler446 perhaps you've spent the most time on this recently, is this your understanding as well?

Reply:
The first one of these wins:

1. mediaType
2. adUnit.bid[]
3. adUnit
4. bid response


if any of those say `backupOnly`, they will be disregarded as long as the bid response has its own renderer.